plugins/ua_config_file_based.c

```diff
@@ -0,0 +1,575 @@
+    *field = out;
+    if(fieldName != NULL) {
+        if(*field)
+            printf("%s: true\n", fieldName);
+        else {
+            printf("%s: false\n", fieldName);
+        }
+    }
+    UA_ByteString_clear(&buf);
+    return UA_STATUSCODE_GOOD;
+}
+PARSE_JSON(DurationField) {
+    UA_Double double_value;
+    DoubleField_parseJson(ctx, fieldName, &double_value, NULL);
+    /* set server config field */
+    UA_Duration *field = (UA_Duration*)configField;
+    *field = (UA_Duration)double_value;
+    return UA_STATUSCODE_GOOD;
+}
+PARSE_JSON(DurationRangeField) {
+    if(fieldName != NULL)
+        printf("%s:\n{\n", fieldName);
+    UA_DurationRange *field = (UA_DurationRange*)configField;
+    cj5_token tok = ctx->tokens[++ctx->index]; //Object token
+    for(size_t j = tok.size/2; j > 0; j--) {
+        tok = ctx->tokens[++ctx->index]; // Key token (string)
+        switch (tok.type) {
+        case CJ5_TOKEN_STRING: {
+            char field_str[tok.size + 1];
+            unsigned int str_len = 0;
+            cj5_get_str(&ctx->result, (unsigned int)ctx->index, field_str, &str_len);
+            if(strcmp(field_str, "min") == 0)
+                parseJsonJumpTable[UA_SERVERCONFIGFIELD_DURATION](ctx, "Min", &field->min, NULL);
+            else if(strcmp(field_str, "max") == 0)
+                parseJsonJumpTable[UA_SERVERCONFIGFIELD_DURATION](ctx, "Max", &field->max, NULL);
+            else {
+                UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Unknown field name.");
+            }
+            break;
+        }
+        default:
+            break;
+        }
+    }
+    /* set server config field */
+    printf("}\n");
+    return UA_STATUSCODE_GOOD;
+}
+PARSE_JSON(UInt32RangeField) {
+    if(fieldName != NULL)
+        printf("%s:\n{\n", fieldName);
+    UA_UInt32Range *field = (UA_UInt32Range*)configField;
+    cj5_token tok = ctx->tokens[++ctx->index]; //Object token
+    for(size_t j = tok.size/2; j > 0; j--) {
+        tok = ctx->tokens[++ctx->index]; // Key token (string)
+        switch (tok.type) {
+        case CJ5_TOKEN_STRING: {
+            char field_str[tok.size + 1];
+            unsigned int str_len = 0;
+            cj5_get_str(&ctx->result, (unsigned int)ctx->index, field_str, &str_len);
+            if(strcmp(field_str, "min") == 0)
+                parseJsonJumpTable[UA_SERVERCONFIGFIELD_UINT32](ctx, "Min", &field->min, NULL);
+            else if(strcmp(field_str, "max") == 0)
+                parseJsonJumpTable[UA_SERVERCONFIGFIELD_UINT32](ctx, "Max", &field->max, NULL);
+            else {
+                UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Unknown field name.");
+            }
+            break;
+        }
+        default:
+            break;
+        }
+    }
+    /* set server config field */
+    printf("}\n");
+    return UA_STATUSCODE_GOOD;
+}
+
+/*----------------------Advanced Types------------------------*/
+PARSE_JSON(BuildInfo) {
```

Comment:
There already is a Buildinfo data type.
You can parse the entire thing in one go via UA_parseJSON.
No need to manually iterate over the members.